File: C Program/Step_9_Functions_Basic.c
#include <stdio.h> 

int FunctionName(){
    
    //That's why we used int like the main function
    //Try Printing Something Here And See What Happens

    return 0;
}

void Void_Function(){

    printf("This is a void type function");
    printf("Meaning it can't return anything, but can print or use other function's returned elements");

}


const char* Hello(){

    return "Hello";
}

//The Functions above does not take any arguments.
//Don't know what arguments are?
//Don't Worry, let's check that out.

void Greet(char* argument){

    printf("%s %s! How are you?", Hello(), argument); //We can call functions in other functions as well
}

//The function above took 1 argument

int main(int count, ...){ 



    FunctionName(); // The functions are called here
    Void_Function();
    
    char name[10];
    printf("\nEnter Your Name: ");
    scanf("%s", name);
    Greet(name);


    return 0;
}


// Now can you make a function that can add two integer values and return the value?

File: C#/Step_9_Functions_Basic.cs
using System;

class Program
{
    static void FunctionName() //This function takes no argument.
    {
        Console.WriteLine("This is a simple function");
    }

    static void SecondFunction(int Argument) {//Void type functions can only print and use Arguments from other functions.
        Console.WriteLine($"This is a function that takes 1 integer argument which is: {Argument}");
    }
    static void Nothing()
    {
        Console.WriteLine("This function can call the Greet function");
        string name = "Botmon";
        Console.WriteLine(Greet(name));
    }

    static string Greet(string name)//Only the functions that are not void type can return.
    {
        return $"Hello {name}! How are you?";
    }
    static void Main()
    {
        FunctionName(); //We code openly in python, but in C# take this Main function as that open canvas. So we call most functions here like this.
        SecondFunction(10);
        Console.Write("Please enter your name: ");
        string name = Console.ReadLine();
        string k = Greet(name);
        Console.WriteLine(k);
        Nothing();
    }
}